### PR TITLE
Add Getting started documentation for otel-cpp sdk

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

Currently, the otel-cpp documentation ( https://opentelemetry-cpp.readthedocs.io/en/latest/ ) contains reference api documentation for both sdk and api and `Getting started` doc for api. But the `Getting started` doc for sdk is missing.
This PR adds the sdk doc. This is how the sdk doc looks like:
https://labhas-opentelemetry-cpp.readthedocs.io/en/latest/sdk/GettingStarted.html

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: docs/public/sdk/GettingStarted.rst [hunk 1]
```diff
@@ -1,4 +1,208 @@
...
+Exporter
+^^^^^^^^
+
+An exporter is responsible for sending the telemetry data to a particular backend. 
+OpenTelemetry offers six tracing exporters out of the box:
+
+- In-Memory Exporter: keeps the data in memory, useful for debugging.
+- Jaeger Exporter: prepares and sends the collected telemetry data to a Jaeger backend via UDP and HTTP.
+- Zipkin Exporter: prepares and sends the collected telemetry data to a Zipkin backend via the Zipkin APIs.
+- Logging Exporter: saves the telemetry data into log streams
```
Comment (ThomsonTan) [inline docs/public/sdk/GettingStarted.rst:18]: nit: ending `.`? The same for line 20.

Comment (lalitb) [reply docs/public/sdk/GettingStarted.rst]: thanks, t was missing at few other places, fixed all.

File: docs/public/sdk/GettingStarted.rst [hunk 2]
```diff
@@ -1,4 +1,208 @@
...
+    opts.service_name = "default_service" ;
+    auto zipkin_exporter =
+        std::unique_ptr<sdktrace::SpanExporter>(new opentelemetry::exporter::zipkin::ZipkinExporter(opts));
+
+    // Jaeger UDP exporter
+    opentelemetry::exporter::jaeger::JaegerExporterOptions opts;
+    opts.server_addr = "localhost";
+    opts.server_port =  6831;
+    auto jaeger_udp_exporter =
+        std::unique_ptr<sdktrace::SpanExporter>(new opentelemetry::exporter::jaeger::JaegerExporter(opts));
```
Comment (ThomsonTan) [inline docs/public/sdk/GettingStarted.rst:44]: Add Jaeger HTTP example?

File: docs/public/sdk/GettingStarted.rst [hunk 3]
```diff
@@ -1,4 +1,208 @@
...
+Span Processor is initialised with an Exporter. Different Span Processors are offered by OpenTelemetry C++ SDK:
+
+- SimpleSpanProcessor: immediately forwards ended spans to the exporter.
+- BatchSpanProcessor: batches the ended spans and send them to exporter in bulk.
+- MultiSpanProcessor: Allows multiple span processors to be active and configured at the same time.
+
+.. code:: cpp
+
+    // simple processor
+    auto simple_processor = std::unique_ptr<sdktrace::SpanProcessor>(
```
Comment (ThomsonTan) [inline docs/public/sdk/GettingStarted.rst:73]: List the full definition of `stdtrace` in the beginning may help the users who follow these example code?

File: docs/public/sdk/GettingStarted.rst (hunk 2 above)
Comment (lalitb) [reply docs/public/sdk/GettingStarted.rst:47]: Yes missed that. Added now.

File: docs/public/sdk/GettingStarted.rst (hunk 3 above)
Comment (lalitb) [reply docs/public/sdk/GettingStarted.rst:86]: Good point. Added it at the beginning.

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**